pkg/providers/k8s/k8s_test.go

```diff
@@ -0,0 +1,208 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/helmfile/vals/pkg/config"
+	"github.com/helmfile/vals/pkg/log"
+)
+
+// Setup:
+// create a local Kubernetes cluster using minikube:
+//   minikube start
+// create a namespace:
+//   kubectl create namespace test-namespace
+// create a secret:
+//   kubectl create secret generic mysecret -n test-namespace --from-literal=key=p4ssw0rd
+
+func Test_getObject(t *testing.T) {
+	homeDir, _ := os.UserHomeDir()
+	testcases := []struct {
+		namespace      string
+		name           string
+		kubeConfigPath string
+		want           map[string][]uint8
+		wantErr        string
+	}{
+		{
+			namespace:      "test-namespace",
+			name:           "mysecret",
+			kubeConfigPath: fmt.Sprintf("%s/.kube/config", homeDir),
+			want:           map[string][]uint8{"key": []uint8("p4ssw0rd")},
+			wantErr:        "",
+		},
+		// kubeConfigPath does not exist
+		{
+			namespace:      "test-namespace",
+			name:           "mysecret",
+			kubeConfigPath: "/tmp/does-not-exist",
+			want:           nil,
+			wantErr:        "Unable to build Kubeconfig from vals configuration: stat /tmp/does-not-exist: no such file or directory",
+		},
+		// namespace does not exist
+		{
+			namespace:      "non-existent-namespace",
+			name:           "mysecret",
+			kubeConfigPath: fmt.Sprintf("%s/.kube/config", homeDir),
+			want:           nil,
+			wantErr:        "Unable to get the object from Kubernetes: secrets \"mysecret\" not found",
+		},
+		// secret does not exist
+		{
+			namespace:      "test-namespace",
+			name:           "non-existent-secret",
+			kubeConfigPath: fmt.Sprintf("%s/.kube/config", homeDir),
+			want:           nil,
+			wantErr:        "Unable to get the object from Kubernetes: secrets \"non-existent-secret\" not found",
+		},
+	}
+
+	for i := range testcases {
+		tc := testcases[i]
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			got, err := getObject(tc.namespace, tc.name, tc.kubeConfigPath, "", context.Background())
+			if err != nil {
+				if err.Error() != tc.wantErr {
+					t.Fatalf("unexpected error: want %q, got %q", tc.wantErr, err.Error())
+				}
+			} else {
+				if tc.wantErr != "" {
+					t.Fatalf("expected error did not occur: want %q, got none", tc.wantErr)
+				}
+			}
+
+			if diff := cmp.Diff(tc.want, got); diff != "" {
+				t.Errorf("unexpected result: -(want), +(got)\n%s", diff)
+			}
+		})
+	}
+}
+
+func Test_getKubeConfig(t *testing.T) {
```

Comment:
@bersalazar I thinks this can't cover all case. WDYT?

Reply:
@bersalazar you can try to use AI to help you generate unit tests. it's so cool.

Reply:
You are correct, tests didn't cover KUBECONFIG env var. I've added a few more tests. Can you please review?